### PR TITLE
Do not deploy when GIT_NAME is not defined.

### PR DESCRIPTION
**Pull request by arstrube**
This should stop deployment when Travis is run on forks.

### CODE REVIEW

File: spec/travis_github_deployer_spec.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@
   end
   
   it "will not deploy on a pull request" do
-    ENV['TRAVIS_PULL_REQUEST']="10"
+    ENV['TRAVIS_PULL_REQUEST']="10" or ENV['GIT_NAME']="false"
```
Comment (basvodde) [inline spec/travis_github_deployer_spec.rb]: You know you are just changing the test, right?

Comment (arstrube) [inline spec/travis_github_deployer_spec.rb]: Yeah. That was the intention, and it should not deploy on a fork right?

But the message would be off.. So it should probably be an additional "paragraph" with the appropriate message....

Comment (basvodde) [inline spec/travis_github_deployer_spec.rb]: I don't understand. Adding a "or" in the spec will not change any behavior...

Comment (arstrube) [inline spec/travis_github_deployer_spec.rb]: I admit I'm lost. Could you please check the snippet I put up in the pull request. If these tests don't change behavior, then what does?

Comment (arstrube) [inline spec/travis_github_deployer_spec.rb]: In plain English (so I won't get tripped up by Ruby stuff):

Current behavior: "if the environment variable TRAVIS_PULL_REQUEST is set, do not deploy and output a message to that effect."

New additional behavior: "if the environment variable GIT_NAME is NOT set, do not deploy and output a message to the effect that we won't deploy on a fork"

I dont seem to be able to translate a simple statement like that into Ruby correctly :-(

Comment (basvodde) [inline spec/travis_github_deployer_spec.rb]: Ok. Why GIT_NAME is not set too then?

The code you submitted doesn't make much sense as you changed an assignment. In C++ it would look like:

a = 10 || b = 20

And b = 20 will never be executed as a = 10 is always true. Plus you only changed the tests and not the code (or at least the pull request only contains changes in the test).

What case will GIT_NAME be empty and TRAVIS_PULL_REQUEST != 10 ?

Comment (arstrube) [inline spec/travis_github_deployer_spec.rb]: Ookay. Even if this sounds really stupid -- if this is the TEST then where is the CODE?

P.S. I was wondering about the '=' but since I don't know Ruby..... ;-P

Sorry, I was trying to be helpful. So I now know how to write a test for what I want (probably the post in the issue (not pull request, sorry) is a passable test, actually) but where does the code go?

Comment (basvodde) [inline spec/travis_github_deployer_spec.rb]: Its fine :) Working in a new language isn't easy and I think its good you dare to look at it ;P Most people wouldn't.

The code is in the lib directory...

File: spec/travis_github_deployer_spec.rb [hunk 2]
```diff
@@ -6,13 +6,15 @@
   subject { TravisGithubDeployer.new}
     
   before(:each) do
-    @git = mock
+    ENV::clear
```
Comment (arstrube) [inline spec/travis_github_deployer_spec.rb:9]: Environment needs to be explicitly cleared, or values from previous test cases will still be around.

File: spec/travis_github_deployer_spec.rb [hunk 3]
```diff
@@ -6,13 +6,15 @@
...
+    @git = double
```
Comment (arstrube) [inline spec/travis_github_deployer_spec.rb:10]: mock is out of action. That was a real headache to fix.

File: spec/travis_github_deployer_spec.rb [hunk 4]
```diff
@@ -22,12 +24,19 @@
     subject.deploy
   end
   
-  it "will not deploy on a pull request" do
+    it "will not deploy on a pull request" do
```
Comment (arstrube) [inline spec/travis_github_deployer_spec.rb]: Oops. this indent is bad. Need to tix it.

File: spec/travis_github_deployer_spec.rb [hunk 5]
```diff
@@ -58,11 +67,6 @@
       subject.set_username_based_on_environment_variable    
     end
       
-    it "Should give an error message when the GIT_NAME isn't set" do
```
Comment (arstrube) [inline spec/travis_github_deployer_spec.rb:61]: This can no longer happen.

File: spec/travis_github_deployer_spec.rb (hunk 2 above)
Comment (basvodde) [inline spec/travis_github_deployer_spec.rb:9]: Ah yeah, nice one.

**Comment by basvodde** [comment]
Ok, change makes a lot more sense now :) Ready to merge?

**Comment by arstrube** [comment]
As ready as I can be :)
